cargo-dist/src/tasks.rs

```diff
@@ -1888,6 +1888,19 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
                 }
             }
 
+            if target.ends_with("linux-musl")
+                && self.inner.tools.cargo.host_target.ends_with("linux-gnu")
+            {
```

Comment:
hmm, any particular motivation to dropping the "target != host_target" check? (Not sure if it would ever trip, but would it ever be wrong..?)

I was thinking we might want to simplify this + macos down to just "always run rustup if we're not building for host", but that might be too far..?

Reply:
Just figure that since this check implicitly contains that check, there's no reason to add the extra conditional.